src/expr/src/expr/expr_unary.rs

```diff
@@ -75,6 +75,7 @@ macro_rules! gen_cast {
 
             { varchar, date, str_to_date },
             { varchar, time, str_to_time },
+            { varchar, interval, str_parse },
             { varchar, timestamp, str_to_timestamp },
             { varchar, timestampz, str_to_timestampz },
```

Comment:
* `date`, `time`, `timestamp` could also use `FromStr` and `parse`.
* `timestampz` would need a wrapper type.
* `bool` means we will need our own trait `PgInput` rather than builtin `FromStr`

As a result, I have not decided which solution is good.